### PR TITLE
Add extra_headers option to get methods

### LINKED ISSUE #193: Add `extra_headers` option to HTTP methods in GitHubAPI
**Problem:** Currently, there is no straightforward way to pass custom headers when making requests with the `GitHubAPI` class. This limits the ability to use certain API features or endpoints that require specific headers to be set (such as [these](https://github.com/PyGithub/PyGithub/blob/300c5015d41236eb410b9cbe895258294e3500af/github/Consts.py#L58-L133)).

**Proposed solution:** I propose adding an `extra_headers` parameter to the methods in the `GitHubAPI` class. This optional parameter would accept a dictionary of custom headers that can be passed when making requests. This would allow users to easily set custom headers for their requests as needed.

For example, to fetch stargazers with their corresponding dates:

```python
async def get_stargazers_with_dates(full_repo_name):
    headers = {"Accept": "application/vnd.github.v3.star+json"}
    async with httpx.AsyncClient() as client:
        gh = gidgethub.httpx.GitHubAPI(client, "basnijholt", oauth_token=token)
        repo_owner, repo_name = full_repo_name.split("/")
        stats_contributors = await gh.getitem(
            f"/repos/{repo_owner}/{repo_name}/stargazers", extra_headers=headers
        )
        return stats_contributors

await get_stargazers_with_dates("basnijholt/adaptive-lighting")
```

**Implementation:** I have opened a PR (#192) with the proposed changes.

### PR DESCRIPTION
**Pull request by basnijholt**
This allows to do for example:

```python
async def get_stargazers_with_dates(full_repo_name):
    headers = {"Accept": "application/vnd.github.v3.star+json"}
    async with httpx.AsyncClient() as client:
        gh = gidgethub.httpx.GitHubAPI(client, "basnijholt", oauth_token=token)
        repo_owner, repo_name = full_repo_name.split("/")
        stats_contributors = await gh.getitem(
            f"/repos/{repo_owner}/{repo_name}/stargazers", extra_headers=headers
        )
        return stats_contributors

await get_stargazers_with_dates("basnijholt/adaptive-lighting")
```

Some more extra headers folks might want to use:

https://github.com/PyGithub/PyGithub/blob/300c5015d41236eb410b9cbe895258294e3500af/github/Consts.py#L58-L133

Closes #193

### CODE REVIEW

**Comment by brettcannon** [comment]
Would you mind opening a corresponding issue so we can discuss the API proposal there?

**Comment by brettcannon** [comment]
And without doing a review, I noticed there are no tests.

**Comment by basnijholt** [comment]
I have opened #193, I can add some tests once we settle on an implementation :-)

**Comment by basnijholt** [comment]
I have opened #193 and I added some tests.

File: tests/test_abc.py [hunk 1]
```diff
@@ -255,6 +255,31 @@ async def test_getiter(self):
...
+        ):
+            data.append(item)
+        assert gh.method == "GET"
+        assert gh.headers["X-Custom-Header"] == "custom_value"
+        assert len(data) == 4
+        assert data[0] == 1
+        assert data[1] == 2
+        assert data[2] == 1
+        assert data[3] == 2
+
```
Comment (basnijholt) [inline tests/test_abc.py:282]: I could also just add 2 extra lines to `test_getiter` above instead of this whole test.

**Comment by basnijholt** [comment]
I believe the CI failure is unrelated to these changes.

**Comment by Mariatta** [comment]
Unrelated but still needs to be fixed before merging this PR. Perhaps we just need to bump the sphinx dependency. It can be done as a separate PR.

**Comment by basnijholt** [comment]
I believe I have fixed the Sphinx issue in https://github.com/brettcannon/gidgethub/pull/192/commits/4cd6ead3176c3b156caddbedfe72756a97804eb7.

**Comment by Mariatta** [comment]
Thank you! Can you also update the changelog

**Comment by Mariatta** [comment]
and documentation to document the new extra_headers parameter

**Comment by basnijholt** [comment]
@Mariatta, I have
- fixed the conflicts
- added documentation
- updated the changelog

Thanks for reviewing!

**Comment by basnijholt** [comment]
We’re now running into https://github.com/urllib3/urllib3/issues/2168

File: .readthedocs.yaml [hunk 2]
```diff
@@ -11,6 +11,6 @@ sphinx:
 
 # Optionally set the version of Python and requirements required to build your docs
 python:
-   version: 3.7
+   version: 3.11
```
Comment (Mariatta) [inline .readthedocs.yaml:14]: I think you need to set `os: ubuntu- 22.04`
https://docs.readthedocs.io/page/config-file/v2.html

**Review by Mariatta — CHANGES_REQUESTED** [review]
The readthedocs is still failing.

File: .readthedocs.yaml (hunk 2 above)
Comment (Mariatta) [reply .readthedocs.yaml:14]: So maybe the file should look more like this:

```
build:
  os: ubuntu-22.04
  tools:
    python: "3.11"


python:
   install:
   - requirements: docs_requirements.txt
   
```

Comment (basnijholt) [reply .readthedocs.yaml]: Thanks!

**Comment by basnijholt** [comment]
The docs build successfully now :tada:

https://gidgethub--192.org.readthedocs.build/en/192/abc.html

**Review by Mariatta — APPROVED** [review]
Thanks!

**Review by brettcannon — APPROVED** [review]
I personally would like to set `{}` as the default value since it isn't modified, but I also know using a mutable default value makes people uncomfortable, so I'm fine with `None` as a default.

**Comment by brettcannon** [comment]
Thanks!

**Pull request merged**